flanker/addresslib/address.py

```diff
@@ -390,27 +547,61 @@ def supports_routing(self):
 
     def full_spec(self):
```

Comment:
I would suggest make `full_spec` raise exception if producing ASCII is not possible, and getting rid of `to_ace`. In this case, otherwise the type of returned value maybe wither str or unicode, and that ambiguity may be a source of problems.